### PR TITLE
BaseLoss cleanup

### PR DESCRIPTION
**Pull request by tjlane**
This PR targets the `BaseLoss` class. The aim was to:
* simplify the abstraction
* correctly manage state in an object-oriented way
* modify the downstream classes in `mse.py` to conform
* simplify the `mse` losses by splitting the `Loss`  class into two:
  * one for coordinates
  * one for structures, whose coordinates need to be sliced
* apply stringing linting/type-checking

I had to bump python to `3.11` to gain access to the lovely `StrEnum` class.

This PR is too big already. In a future PR I will:
* put things under unit tests

### CODE REVIEW

File: losslab/losses/mse.py [hunk 1]
```diff
@@ -2,139 +2,89 @@
 
 from __future__ import annotations
 
-from typing import Any
-
 import numpy as np
 import torch
+from SFC_Torch.io import PDBParser
```
Comment (CKalicki) [inline losslab/losses/mse.py:7]: why use this pdb parser -- should there be an internal one? and for CIFs

Comment (tjlane) [reply losslab/losses/mse.py:7]: @CKalicki that's a good point, and something that might change longer term. Right now, downstream code relies on SFcalculator's PDB class. Changing it would be another discussion and PR. @minhuanli if you have input, speak up!

Comment (minhuanli) [reply losslab/losses/mse.py:7]: The `PDBParser` in `SFC_Torch` is the necessary class for working with SFC. I originally developed it because I couldn't find an existing PDB class that effectively handles space groups, unit-cell sizes, B-factors, and occupancy while remaining compatible with `gemmi` selection syntax and providing a flat, non-hierarchical array representation of model properties.

I am currently testing mmCIF support, which is a work in progress. You can track the updates here: https://github.com/rs-station/SFcalculator_torch/pull/25.

What specific properties are you looking for in an internal PDB class? Since SFC is a core part of RSS and many LossLab reciprocal loss functions will depend on it, keeping the architecture modular seems not bad?

Comment (tjlane) [reply losslab/losses/mse.py:7]: this might be a good conversation but it probably belongs somewhere else -- I'll consider this a fixed decision for the immediate purpose of this PR

Comment (CKalicki) [reply losslab/losses/mse.py:7]: sounds gucci

**Review by CKalicki — APPROVED** [review]
